BraveWallet/Crypto/BuySwapSwap/SendTokenView.swift

```diff
@@ -71,6 +80,15 @@ struct SendTokenView: View {
                                                      sendTokenStore.selectedSendToken?.symbol ?? ""),
                     text: $amountInput
           )
+            .onChange(of: amountInput, perform: { value in
+              guard let token = sendTokenStore.selectedSendToken else { return }
+              // will ignore if input is not a decimal number
+              if let decimal = Decimal(string: value), -decimal.exponent > 0 {
```

Comment:
should i use BDecimal?

Reply:
Yeah you should consider using `BDouble` or just toss values into `WeiFormatter` since we need to support not only 18 digits _after_ the decimal point but all the data before it